### PR TITLE
Add E2E tests for Elastic Stack 8.0.0-SNAPSHOT

### LINKED ISSUE #2608: Add E2E tests for Elastic Stack 8.0.0-SNAPSHOT
This depends on https://github.com/elastic/cloud-on-k8s/issues/2607.

### PR DESCRIPTION
**Pull request by thbkrkr**
Resolved #2608.

### CODE REVIEW

**Comment by thbkrkr** [comment]
I saw instabilities on EnterpriseSearch and License tests but it's a bit painful to repro so I don't know if it's flakyness or a real bug.

Proposal: create a separate job only for the 8.0.0-SNAPSHOT with notifications turned off in order not to add noise until it seems stable enough.

**Comment by thbkrkr** [comment]
The last commit creates a new pipeline, duplicated from `e2e-tests-snapshot-versions-gke`, to run E2E tests with Elastic Stack 8.0.0-SNAPSHOT with notifications disabled by default.

**Comment by thbkrkr** [comment]
jenkins test this please

File: .ci/jobs/e2e-tests-snapshot-versions-8x-gke.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
+- job:
+    description: Run ECK E2E tests on GKE every night to verify ECK against Elastic Stack dev 8.x versions
+    name: cloud-on-k8s-e2e-tests-snapshot-versions
```
Comment (barkbay) [inline .ci/jobs/e2e-tests-snapshot-versions-8x-gke.yml:4]: Should we change the name here ?

Also I'm wondering if it makes sense to have only two jobs and avoid to set the version in the job/file name:

* One for stable versions of the stack : `cloud-on-k8s-e2e-tests-stable-versions` , which would test 6.8 to 7.9 in parallel 
* One for dev/snapshot versions: `cloud-on-k8s-e2e-tests-snapshot-versions`, which would test 7.10 and 8 in parallel 

wdyt ?

Comment (thbkrkr) [reply .ci/jobs/e2e-tests-snapshot-versions-8x-gke.yml]: Yes, for the missing name change.

Interesting idea for the two jobs but I'm afraid I won't have time to delve into this now.

**Review by barkbay — APPROVED** [review]
LGTM !

**Comment by thbkrkr** [comment]
Creates a new pipeline, duplicated from e2e-tests-snapshot-versions-gke,
to run E2E tests with Elastic Stack 8.0.0-SNAPSHOT with notifications disabled by default.

**Pull request merged**